Dockerfile.CI

```diff
@@ -1,43 +1,39 @@
 FROM node:14-alpine as builder-client
 ARG env=production
 ENV npm_config_cache=/tmp/.npm
-RUN mkdir -p /tmp/.npm /app \
-    && chown nobody:nobody /tmp/.npm /app
+RUN mkdir -p /tmp/.npm /app && chown nobody:nobody /tmp/.npm /app
 WORKDIR /app
 USER nobody
-
 ADD git_src/package.json git_src/package-lock.json ./
 RUN npm install
 ADD git_src/webpack.config.cjs git_src/babel.config.cjs ./
 ADD git_src/client ./client
 RUN npm run build-frontend
 
-
 FROM node:14-alpine as builder-app
 ARG env=production
 ENV npm_config_cache=/tmp/.npm
 RUN mkdir -p /app && chown nobody:nobody /app
-COPY --from=builder-client /tmp/.npm /tmp/.npm
+RUN mkdir -p /.npm && chown nobody:nobody /.npm
 WORKDIR /app
-ADD git_src/package.json git_src/package-lock.json ./
+USER nobody
+ADD src ./src
+ADD git_src/package.json git_src/package-lock.json git_src/tsconfig.json ./
 RUN npm install
 RUN npm run build-backend
 
 FROM node:14-alpine
 ARG env=production
 ENV NODE_ENV=${env}
 RUN mkdir -p /app && chown nobody:nobody /app
```

Comment:
unless you have specific requirement, its safer not to give read/write access rights

Reply:
I recommend omitting "&& chown nobody:nobody /app" and only giving the owner access to the subfolder were read & write is happening (if at all).